pkg/kfapp/ossm/ossm_manifests.go

```
@@ -74,6 +86,9 @@ func (o *OssmInstaller) processManifests() error {
 	if err != nil {
 		return internalError(errors.WithStack(err))
 	}
+	if copyFsErr := copyEmbeddedFS(embeddedFiles, "templates", filepath.Join(baseOutputDir, o.Namespace, o.Name)); copyFsErr != nil {
```

I think we should first copy and only then try to load manifests etc. If we fail with copying we can't really do much more, so better to fail fast in my eyes.

On top of that maybe leaving a short comment on why we do it could be helpful.

---

Great call on moving this to be first! Was able to go back to your original, simpler processing methods and that made cleaning things up much easier.